src/transport/CertificateAuthenticatedSessionEstablishment.h

```diff
@@ -0,0 +1,175 @@
+     */
+    CHIP_ERROR WaitForSessionEstablishment(NodeId myNodeId, uint16_t myKeyId, AuthenticatedSessionEstablishmentDelegate * delegate);
+
+    /**
+     * @brief
+     *   Create and send session establishment request using device's operational credentials.
+     *
+     * @param peerAddress      Address of peer to pair
+     * @param myNodeId         Node id of local node
+     * @param peerNodeId       Node id of the peer node
+     * @param myKeyId          Key ID to be assigned to the secure session on the peer node
+     * @param delegate         Callback object
+     *
+     * @return CHIP_ERROR      The result of initialization
+     */
+    CHIP_ERROR EstablishSession(const Transport::PeerAddress peerAddress, NodeId myNodeId, NodeId peerNodeId, uint16_t myKeyId,
+                                AuthenticatedSessionEstablishmentDelegate * delegate);
+
+    /**
+     * @brief
+     *   Derive a secure session from the paired session. The API will return error
+     *   if called before pairing is established.
+     *
+     * @param info        Information string used for key derivation
+     * @param info_len    Length of info string
+     * @param session     Referene to the sescure session that will be
+     *                    initialized once pairing is complete
+     * @return CHIP_ERROR The result of session derivation
+     */
+    virtual CHIP_ERROR DeriveSecureSession(const uint8_t * info, size_t info_len, SecureSession & session);
+
+    /**
+     * @brief
+     *   Handler for peer's messages, exchanged during pairing handshake.
+     *
+     * @param packetHeader Message header for the received message
+     * @param peerAddress  Source of the message
+     * @param msg          Message sent by the peer
+     * @return CHIP_ERROR The result of message processing
+     */
+    virtual CHIP_ERROR HandlePeerMessage(const PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
+                                         System::PacketBufferHandle msg);
+
+    /**
+     * @brief
+     *  Return the associated secure session peer NodeId
+     *
+     * @return Optional<NodeId> The associated peer NodeId
+     */
+    NodeId GetPeerNodeId() const { return mConnectionState.GetPeerNodeId(); }
+
+    /**
+     * @brief
+     *  Return the associated peer key id
+     *
+     * @return uint16_t The associated peer key id
+     */
+    uint16_t GetPeerKeyId() { return mConnectionState.GetPeerKeyID(); }
+
+    /**
+     * @brief
+     *  Return the associated local key id
+     *
+     * @return uint16_t The assocated local key id
+     */
+    uint16_t GetLocalKeyId() { return mConnectionState.GetLocalKeyID(); }
+
+    Transport::PeerConnectionState & PeerConnection() { return mConnectionState; }
+
+private:
+    enum SigmaErrorType : uint8_t
+    {
+        kNoSharedTrustRoots   = 0x01,
+        kNoSharedCipherSuite  = 0x02,
+        kNoSharedNamedCurve   = 0x03,
```

Comment:
`NoSharedCipherSuite` and `NoSharedNamedCurve` are not in the spec, as far as I can tell.

Reply:
Looks like it got removed since I last checked the spec. Removing from the code as well.